applications/MetisApplication/custom_processes/metis_divide_heterogeneous_input_process.h

```diff
@@ -674,21 +682,31 @@ class MetisDivideHeterogeneousInputProcess : public MetisDivideInputToPartitions
               int MajorityPartition = NeighbourPartitions[ FindMax(FoundNeighbours,NeighbourWeights) ];
               {
                   *itPart = MajorityPartition;
-                  PartitionWeights[MajorityPartition]++;
               }
 
               // ensure conditions sharing nodes with an element have same partition as the element
               IO::ConnectivitiesContainerType::value_type tmp(*itCond);
               std::sort(tmp.begin(), tmp.end());
 
-              for (SizeType i=0; i<NumElements; i++)
-              {
-                  if ( std::includes(ElementsSorted[i].begin(), ElementsSorted[i].end(), tmp.begin(), tmp.end()) )
-                  {
-                      *itPart = rElemPartition[i];
-                      break;
+              for (std::vector<SizeType>::const_iterator itNode = itCond->begin(); itNode != itCond->end(); ++itNode) {
+                for(auto shared_element : mNodeConnectivities[*itNode - 1]) {
+                  // Would it be faster to sort the element here as well?
+                  // Should be as far as "numConditions * conPerNode >> numElements", but not otherwise
+                  if ( std::includes(ElementsSorted[shared_element].begin(), ElementsSorted[shared_element].end(), tmp.begin(), tmp.end()) ) {
+                    *itPart = rElemPartition[shared_element];
+                    break;
                   }
+                }
               }
+
+            //   for (SizeType i=0; i<NumElements; i++)
+            //   {
+            //       if ( std::includes(ElementsSorted[i].begin(), ElementsSorted[i].end(), tmp.begin(), tmp.end()) )
+            //       {
+            //           *itPart = rElemPartition[i];
+            //           break;
+            //       }
+            //   }
```

Comment:
can you please remove the commented code before merging?

Reply:
Sure!

Reply:
I can approve afterwards :)